taichi/ui/backends/vulkan/renderables/set_image.cpp

```diff
@@ -154,19 +154,23 @@ void SetImage::update_vertex_buffer_() {
       {{1.f, 1.f, 0.f}, {0.f, 0.f, 1.f}, {1.f, 0.f}, {1.f, 1.f, 1.f}},
       {{1.f, -1.f, 0.f}, {0.f, 0.f, 1.f}, {1.f, 1.f}, {1.f, 1.f, 1.f}},
   };
-
+  // Our actual VBO might only use the first several attributes in `Vertex`,
+  // therefore this slicing & copying for each Vertex.
+  const size_t vbo_size = config_.vbo_size();
   {
-    Vertex *mapped_vbo =
-        (Vertex *)app_context_->device().map(staging_vertex_buffer_);
-
-    memcpy(mapped_vbo, vertices.data(),
-           (size_t)config_.vertices_count * sizeof(Vertex));
+    char *mapped_vbo =
```

Comment:
I remembered why I didn't go with the bitmask approach in the beginning. This part would require the users to always follow the order Vertex attributes are defined. I.e., you can use {pos, uv}, but not {uv, pos}. Guess this limitation is not that significant, though.